src/routes/services/[id]/calcom/start.json.ts

```diff
@@ -0,0 +1,167 @@
+			calcom: {
+				image: `${image}:${version}`,
+				environmentVariables: {
+					BASE_URL: domain,
+					DATABASE_URL: `postgresql://${postgresqlUser}:${postgresqlPassword}@${id}-postgresql:5432/${postgresqlDatabase}?sslmode=disable`,
+					EMAIL_FROM: emailFrom,
+					EMAIL_SERVER_HOST: emailSmtpHost,
+					EMAIL_SERVER_PORT: emailSmtpPort,
+					EMAIL_SERVER_USER: emailSmtpUser,
+					EMAIL_SERVER_PASSWORD: emailSmtpPassword,
+					MS_GRAPH_CLIENT_ID: msGraphClientId,
+					MS_GRAPH_CLIENT_SECRET: msGraphClientSecret,
+					ZOOM_CLIENT_ID: zoomClientId,
+					ZOOM_CLIENT_SECRET: zoomClientSecret,
+					GOOGLE_API_CREDENTIALS: googleApiCredentials,
+					NEXT_PUBLIC_WEBAPP_URL: domain,
+					NEXT_PUBLIC_LICENSE_CONSENT: true,
+					CALCOM_TELEMETRY_DISABLED: 1,
+					CALCOM_LICENSE_KEY: licenseKey,
+					NEXTAUTH_SECRET: `${authSecret.replace(/\$/g, '$$$')}`,
+					CALENDSO_ENCRYPTION_KEY: `${encryptionKey.replace(/\$/g, '$$$')}`
+				}
+			},
+			postgresql: {
+				image: 'postgres:12-alpine',
+				volume: `${id}-postgresql-data:/var/lib/postgresql/data`,
+				environmentVariables: {
+					POSTGRES_USER: postgresqlUser,
+					POSTGRES_PASSWORD: postgresqlPassword,
+					POSTGRES_DB: postgresqlDatabase
+				}
+			}
+		};
+		if (serviceSecret.length > 0) {
+			serviceSecret.forEach((secret) => {
+				config.calcom.environmentVariables[secret.name] = secret.value;
+			});
+		}
+
+		const composeFile: ComposeFile = {
+			version: '3.8',
+			services: {
+				[id]: {
+					container_name: id,
+					image: config.calcom.image,
+					environment: config.calcom.environmentVariables,
+					networks: [network],
+					volumes: [],
+					restart: 'always',
+					labels: makeLabelForServices('calcom'),
+					...(exposePort ? { ports: [`${exposePort}:${port}`] } : {}),
+					deploy: {
+						restart_policy: {
+							condition: 'on-failure',
+							delay: '5s',
+							max_attempts: 3,
+							window: '120s'
+						}
+					},
+					depends_on: [`${id}-postgresql`]
+				},
+				[`${id}-postgresql`]: {
+					image: config.postgresql.image,
+					container_name: `${id}-postgresql`,
+					environment: config.postgresql.environmentVariables,
+					networks: [network],
+					volumes: [config.postgresql.volume],
+					restart: 'always',
+					deploy: {
+						restart_policy: {
+							condition: 'on-failure',
+							delay: '5s',
+							max_attempts: 3,
+							window: '120s'
+						}
+					}
+				}
+			},
+			networks: {
+				[network]: {
+					external: true
+				}
+			},
+			volumes: {
+				[config.postgresql.volume.split(':')[0]]: {
+					name: config.postgresql.volume.split(':')[0]
+				}
+			}
+		};
+		const composeFileDestination = `${workdir}/docker-compose.yaml`;
+		await fs.writeFile(composeFileDestination, yaml.dump(composeFile));
+
+		try {
+			await asyncExecShell(`DOCKER_HOST=${host} docker compose -f ${composeFileDestination} pull`);
+			await asyncExecShell(`DOCKER_HOST=${host} docker compose -f ${composeFileDestination} up -d`);
+			return {
```

Comment:
For the cal.io first user thing, you could look into putting some code here to read if there are any users and if there isn't one add a new one;
This might require adding some new fields to the service config for the initial user's credentials.

Reply:
I actually saw one from a service I could use, the only issue is that I don't know the schema for it

Reply:
I just checked the [schema here](https://github.com/calcom/cal.com/blob/main/packages/prisma/schema.prisma#L122-L181) and it's rather complex, so I'm not even sure if it's a good idea to manually create it with plain SQL.

Reply:
It doesn't look that bad; most of them are options or have defaults.

And you could test with a local version of the container running (outside of Coolify) and pgadmin to get the SQL right.

Reply:
Currently running without docker on my machine (and developing on Codespaces) so that will be a bit complex to setup with stuff I'm already running.

Reply:
It's currently the middle of my work day, so I can't do anything right now 😆
I might have some time to poke around tonight.

Reply:
Yeah if I had the dev environment set up I definitely would have done this step 😄